src/store/slices/register.slice.ts

```diff
@@ -37,52 +37,87 @@ const initialState: RegisterState = {
+}
+
 export const register = createAsyncThunk("register", async (request: RegisterRequest, { rejectWithValue, dispatch, getState }) => {
   try {
-    const putProfileRes = await fetchData.put('/profile/current', request.form);
-    if (putProfileRes.status !== 200) throw { response: putProfileRes };
-
-    const registerSubscriptionRes = await fetchData.post(`/profile/current/subscriptions/${request.tierId}`);
-    if (registerSubscriptionRes.status !== 201) throw { message: registerSubscriptionRes.data };
-    const subscriptionId = registerSubscriptionRes.data.id as string;
-
-    const subscriptionsRes = await fetchData.get('/profile/current/subscriptions');
-    if (subscriptionsRes.status !== 200) throw { message: subscriptionsRes.data };
-    const subscription = subscriptionsRes.data.find((item: any) => item.id === subscriptionId);
-
-    if (!subscription) throw { message: 'There\'s no subscription registered' };
-    if (subscription.status !== 'pending') throw { message: 'The subscription it\'s not pending' };
-
-    const subscriptionPrice = BigNum.from_str(subscription.price.toString());
-    const balanceRes = await fetchData.get('/profile/current/balance');
-    if (balanceRes.status !== 200) throw { message: balanceRes.data };
-    const balance = BigNum.from_str(balanceRes.data.toString());
-
-    let lessBalance = false;
-    if (balance.less_than(subscriptionPrice)) {
-      lessBalance = true;
-    } else {
-      const difference = balance.checked_sub(subscriptionPrice);
-      lessBalance = difference.less_than(BigNum.from_str('0'));
-    }
 
-    if (!lessBalance) throw { message: 'Not enough balance' }; 
+    await updateUserProfile(request.form);
+
+    let transactionId: string|null = null;
+    let subscription = await getUserPendingSubscriptionByTierId(request.tierId);
+
+    if (!subscription) {
+      const subscriptionId = await registerSubscription(request.tierId);
+      subscription = await getUserSubscriptionById(subscriptionId);
+      if (!subscription) throw { message: 'There\'s no subscription registered' };
+      if (subscription.status !== 'pending') throw { message: 'The subscription it\'s not pending' };
 
-    const oneAdaInLovelaces = 1000000;
-    let fee = BigNum.from_str(oneAdaInLovelaces.toString()); // 1 ADA in lovelaces - min req for cardano wallet txn
-    if (fee.less_than(subscriptionPrice)) fee = subscriptionPrice;
+      const subscriptionPrice = BigNum.from_str(subscription.price.toString());
+      const balance = await getUserBalance();
 
-    const { wallet, walletAddress: address, stakeAddress: payer } = (getState() as RootState).auth;
-    const paymentRes = await dispatch(payFromWallet({ wallet, address, fee, payer }));
-    if (paymentRes?.error?.message) throw paymentRes?.error;
-    const transactionId = paymentRes.payload;
+      const { isBalanceEnough, fee } = await calculateFee(subscriptionPrice, balance);
+      if (!isBalanceEnough) throw { message: 'Not enough balance' };
```

Comment:
In this condition, when is the user prompted to actually perform the transaction; in a case where he some balance in his account?